### PR TITLE
docs: automate components overview build

### PR DESCRIPTION
**Pull request by juliodialpad**
## Description

- Added script to get the frontmatter information from '/components/' path.
- Automated the building of /components/ overview.

## Pull Request Checklist

 - [x] Ask the contributors if a similar effort is already in process or has been solved.
 - [x] Review the [contribution guidelines](https://github.com/dialpad/dialtone/blob/staging/.github/CONTRIBUTING.md).
 - [x] Use `staging` as your pull request's base branch. (All PRs using `production` as its base branch will be declined).
 - [x] Ensure all `gulp` scripts successfully compile.
 - [ ] Update, remove, or extend all affected documentation.
 - [x] Ensure no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).

### CODE REVIEW

File: docs/_data/site-nav.json [hunk 1]
```diff
@@ -577,8 +577,8 @@
           "link": "/components/root_layout.html"
         },
         {
-          "text": "Select",
-          "link": "/components/select.html"
+          "text": "Select Menu",
```
Comment (francisrupert) [inline docs/_data/site-nav.json]: Any particular reason this was renamed?

File: docs/components/index.md [hunk 2]
```diff
@@ -4,361 +4,34 @@ desc: Reusable components solving common UI needs, designed and built to be asse
...
-    </div>
-  </router-link>
-  <router-link class="dialtone-wall__item" to="/components/button_group.html">
-    <div class="dialtone-wall__image"><img class="dialtone-wall__thumb" alt="" :src="$withBase('/assets/images/components/button-group.png')"></div>
-    <div class="dialtone-wall__details">
-      <div class="dialtone-wall__title">
-        <span class="dialtone-wall__title-text">Button group</span>
-      </div>
-      <div class="dialtone-wall__description">Button groups are used to group buttons that have a relationship or similar actions.</div>
+  <router-link
```
Comment (francisrupert) [inline docs/components/index.md]: Will need to account for items listed here, but don't yet have a page. Look at https://dialpad.design/design/ for example...

```html

<div class="dialtone-wall__item dialtone-wall__item--disabled"...
```
or
```html

<router-link class="dialtone-wall__item"...

```

File: docs/components/index.md [hunk 3]
```diff
@@ -4,361 +4,34 @@ desc: Reusable components solving common UI needs, designed and built to be asse
...
+    v-for="component in $page.componentsFrontmatter"
+    :key="component.title"
+    class="dialtone-wall__item"
+    :to="`/components/${component.link}.html`">
+    <div class="dialtone-wall__image">
```
Comment (francisrupert) [inline docs/components/index.md]: Suppress if no image available, which currently occurs on https://dialpad.design/design/ and https://dialpad.design/guides/

File: docs/components/index.md [hunk 4]
```diff
@@ -4,361 +4,34 @@ desc: Reusable components solving common UI needs, designed and built to be asse
...
+
+<script setup>
+const badgeKindClass = (status) => {
+  switch(status) {
+    case 'new':
+      return 'd-badge--bulletin';
+    default:
+      return 'd-badge--info'
+  }
+};
```
Comment (francisrupert) [inline docs/components/index.md]: We'll probably have more badge types once I figure out strategy for component health. At least for now, make "Ready" as `d-badge--success` (green).

**Review by francisrupert — CHANGES_REQUESTED** [review]
Loving this so much.

Since you have it tied directly to Commponents `index.md` how do we reuse it on other landing pages? https://dialpad.design/design/ and https://dialpad.design/guides/

Note I previously added a `"planned": true` placeholder in `site-nav.json` as a way to get it in the left nav. So that's redundancy on my part, how can we reconcile? I like that it's in the front matter, can that be reused for the left-nav, or should it solely be in `site-nav.json` or...?

image

File: docs/_data/site-nav.json (hunk 1 above)
Comment (juliodialpad) [reply docs/_data/site-nav.json]: Yes, `Select menu` is the "right" naming, is was called that way in everywhere else except of that file, so I wanted to keep the aligned.

Comment (juliodialpad) [reply docs/_data/site-nav.json]: Dialtone-vue: https://vue.dialpad.design/?path=/story/components-select-menu--default
Dialtone: 
Screenshot 2022-12-08 at 5 11 08 p m

Also in dialtone was called that way, but not on the sidebar and file.

Comment (francisrupert) [reply docs/_data/site-nav.json]: Ah, ok. Meanwhile: sentence case to "Select menu"

File: docs/components/index.md (hunk 2 above)
Comment (juliodialpad) [reply docs/components/index.md]: Sure, I noticed I was doing it the wrong way...

I'm working now on a more universal component to render them all.

File: docs/_data/site-nav.json [hunk 5]
```diff
@@ -28,7 +28,7 @@
         },
         {
           "text": "Getting Started",
```
Comment (francisrupert) [inline docs/_data/site-nav.json]: "Getting started" (sentence case).

File: docs/.vuepress/views/Overview.vue [hunk 6]
```diff
@@ -0,0 +1,88 @@
+<template>
```
Comment (francisrupert) [inline docs/.vuepress/views/Overview.vue:1]: Love this `Overview.vue` for reuse 🤩

File: docs/_data/site-nav.json [hunk 7]
```diff
@@ -28,7 +28,7 @@
...
-          "link": "/guides/getting-started/"
+          "link": "/guides/getting_started/"
```
Comment (francisrupert) [inline docs/_data/site-nav.json]: Any particular reason for changing to underscore? Tad bit of concern on breaking existing links (e.g. bookmarks, etc.). Internal links are thrown around and a lot in Slack, DP, emails, etc and we'll be throwing 404s at everyone.

File: docs/_data/site-nav.json [hunk 8]
```diff
@@ -37,11 +37,11 @@
         },
         {
           "text": "Writing guidelines",
-          "link": "/guides/content/"
+          "link": "/guides/writing_guidelines/"
         },
         {
-          "text": "Accessibility",
-          "link": "/guides/accessibility/"
+          "text": "Accessibility and inclusive design",
```
Comment (francisrupert) [inline docs/_data/site-nav.json]: Let's change this to just "Accessibility." 

image

**Review by francisrupert — CHANGES_REQUESTED** [review]

File: docs/_data/site-nav.json (hunk 7 above)
Comment (juliodialpad) [reply docs/_data/site-nav.json]: Had to change some links...

Now that we are building some stuff automatically (sidebar, prev and next links, overview) We need to have a standard naming convention so I needed to decide if we wanted to stay with underscore or score case, so I decided to stay with "score-case" but many links were not conforming with that so they had to be renamed.

File: docs/_data/site-nav.json (hunk 8 above)
Comment (juliodialpad) [reply docs/_data/site-nav.json]: Renamed but added a "hack" to maintain it that way...

The naming in the sidebar, must match the Title in the frontmatter and the url... so we need to add a `shortTitle` property to the frontmatter to get around this.

File: docs/guides/content/index.md [hunk 9]
```diff
@@ -1,5 +1,6 @@
 ---
 title: Writing guidelines
+shortTitle: content
```
Comment (francisrupert) [inline docs/guides/content/index.md]: Like the `shortTitle` approach, though keep this one as "Writing guidelines".

**Review by francisrupert — APPROVED** [review]
Great updates. Approved with one tiny change request.

File: docs/guides/content/index.md (hunk 9 above)
Comment (juliodialpad) [reply docs/guides/content/index.md]: Renamed

**Comment by juliodialpad** [comment]
@braddialpad any comment on the technical side here? or is it good to go?

**Comment by braddialpad** [comment]
Having a look

**Review by braddialpad — APPROVED** [review]
Very nice thanks

**Pull request merged**